java/ql/src/semmle/code/java/dataflow/internal/DataFlowImpl.qll

```diff
@@ -748,6 +748,12 @@ private module Stage2 {
     ApNil() { this = false }
   }
 
+  bindingset[tc, tail]
+  private Ap apCons(TypedContent tc, Ap tail) { result = true and exists(tc) and exists(tail) }
+
+  pragma[inline]
+  private Content getHeadContent(Ap ap) { exists(result) and ap = true }
```

Comment:
That would correspond to a `pop` of an `Ap` just to get the head, so we'd introduce a needless fan-out, which we would then hope that the compiler could eliminate in all stages. Besides this risk I think it would be more confusing, so I don't think it's worth it just to eliminate one predicate-parameter.

Reply:
OK